code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,486 @@
+		return TRUE
+
+/obj/machinery/ftl_shunt/core/attack_hand(var/mob/user)
+	if(sabotaged)
+		var/skill_delay = user.skill_delay_mult(SKILL_ENGINES, 0.3)
+		if(!user.skill_check(SKILL_ENGINES, SKILL_BASIC))
+			to_chat(user, SPAN_DANGER("You are nowhere near experienced enough to stick your hand into that thing."))
+			return FALSE
+		to_chat(user, SPAN_NOTICE("You reach your hand inside of [src] and slowly begin to remove a foreign object ..."))
+		if(!do_after(user, (4 SECOND * skill_delay), src))
+			return FALSE
+		var/obj/item/stack/telecrystal/TC = new
+		TC.amount = sabotaged_amt
+		TC.forceMove(get_turf(user))
+		user.put_in_hands(TC)
+		to_chat(user, SPAN_NOTICE("You remove the object from [src], and it seems to return to normal."))
+		sabotaged = null
+		sabotaged_amt = 0
+		return TRUE
+
+//Custom procs.
+//Finds fuel ports.
+/obj/machinery/ftl_shunt/core/proc/find_ports()
+	var/datum/extension/local_network_member/network = get_extension(src, /datum/extension/local_network_member)
+	var/datum/local_network/lan = network.get_local_network()
+	if(lan)
+		var/list/ports = lan.get_devices(/obj/machinery/ftl_shunt/fuel_port)
+		for(var/obj/machinery/ftl_shunt/fuel_port/FP in ports)
+			if(!FP.master)
+				FP.master = src
+				fuel_ports += FP
+
+//Starts the teleport process, returns 1-6, with 6 being the all-clear.
+/obj/machinery/ftl_shunt/core/proc/start_shunt()
+	var/shunt_distance
+	var/vessel_mass = ftl_computer.linked.get_vessel_mass()
+	var/shunt_turf = locate(shunt_x, shunt_y, GLOB.using_map.overmap_z)
+
+	if(stat & (BROKEN))
+		return FTL_START_FAILURE_BROKEN
+	if(stat & (NOPOWER))
+		return FTL_START_FAILURE_POWER
+
+	if(!length(fuel_ports)) //no fuel ports
+		find_ports()
+		if(!length(fuel_ports))
+			return FTL_START_FAILURE_OTHER
+
+	if(world.time <= cooldown)
+		return FTL_START_FAILURE_COOLDOWN
+
+	shunt_distance = get_dist(get_turf(ftl_computer.linked), shunt_turf)
+	required_fuel_joules = (vessel_mass * JOULES_PER_TON) * shunt_distance
+
+	if(required_fuel_joules > get_fuel(fuel_ports))
+		return FTL_START_FAILURE_FUEL
+
+	charge_time = world.time + get_charge_time()
+	//If we've gotten to this point then we're okay to start charging up.
+	charging = TRUE
+	charge_started = world.time
+
+	var/eta = round((get_charge_time() / 600))
+
+	var/announcetxt = replacetext(shunt_start_text, "%%TIME%%", "[eta] minutes.")
+
+	ftl_announcement.Announce(announcetxt, "FTL Shunt Management System", new_sound = sound('sound/misc/notice2.ogg'))
+	return FTL_START_CONFIRMED
+
+//Cancels the in-progress shunt.
+/obj/machinery/ftl_shunt/core/proc/cancel_shunt(var/silent = FALSE)
+	if(!charging) //Not preparing for a jump.
+		return
+	charging = FALSE
+	charge_time = null
+	charge_started = null
+	cooldown = null
+	required_fuel_joules = null
+	if(silent)
+		ftl_announcement.Announce(shunt_cancel_text, "FTL Shunt Management System", new_sound = sound('sound/misc/notice2.ogg'))
```

Comment:
Confusing usage of this arg; do you mean `!silent`?

Reply:
No, silent is FALSE by default.